### PR TITLE
Bug Fix: Issue#4: The gulp process should not exit now

### PR DESCRIPTION
**Pull request by krshubham**
[Issue 4](https://github.com/Grasia/teem/issues/4)

The basic idea was to use an error handler with each pipe and emit 'end' event to the corresponding pipe.

### CODE REVIEW

**Comment by atfornes** [comment]
Thanks @krshubham, could you please review the indentation and trailing spaces using our jshint rules in your editor?

Using Atom editor is really easy to setup the editor to have automatic indentation.

File: gulpfile.js [hunk 1]
```diff
@@ -221,11 +221,25 @@ var gulp           = require('gulp'),
...
+  console.log(err);
+  this.emit('end');
+}
+
 /*================================================
 =            Report Errors to Console            =
 ================================================*/
 
-gulp.on('error', function(e) {
+gulp.on('error', function(e) {
```
Comment (atfornes) [inline gulpfile.js]: please, erase trailing white spaces. Using a good linter that uses our jshint rules is the preferred way of doing it.

File: gulpfile.js [hunk 2]
```diff
@@ -295,10 +312,12 @@ gulp.task('images', function () {
...
       svgoPlugins: [{removeViewBox: false}],
       use: [pngcrush()]
-    }));
+    }))
+    .on('error', endErrorProcess);
   }
 
-  return stream.pipe(gulp.dest(path.join(config.dest, 'images')));
+  return stream.pipe(gulp.dest(path.join(config.dest, 'images')))
+  .on('error', endErrorProcess);
```
Comment (atfornes) [inline gulpfile.js]: I believe this .on should be indented two spaces towards the right. Please review that it follows our jshint rules.

File: gulpfile.js [hunk 3]
```diff
@@ -530,7 +579,9 @@ function buildManifest (env) {
       exclude: 'app.manifest',
       hash: true
     }))
-  .pipe(gulp.dest(config.dest));
+    .on('error', endErrorProcess)
+  .pipe(gulp.dest(config.dest))
```
Comment (atfornes) [inline gulpfile.js]: please review indentation

Comment (krshubham) [inline gulpfile.js]: Sorry for this, I'll correct this and keep in mind from next time. Also, I'll for sure configure my text editor or use atom as you said its easy to get it configured. 👍

Comment (atfornes) [inline gulpfile.js]: don't worry :) just sharing our formatting tools and rules

**Comment by atfornes** [comment]
@krshubham, can you please open another pull request for the popover?

In addition, please check that:
1. you have jslint and jshint enabled in your editor and that your new code comply with our rules
2. Please do not add in your commit indentation changes in lines that you have not modified.

Thanks!

**Comment by krshubham** [comment]
@atfornes  I am closing this right now and will again make a PR as I messed up in this! 😢

**Pull request closed without merging**